File: CS50x/lecture6/sentimental-readability/readability.py
text = input("Text: ").upper()
letter = 0
space = 1
sentences = 0
for i in range(len(text)):
    if text[i] >= "A" and text[i] <= "Z":
        letter = letter + 1
    if text[i] == " ":
        space = space + 1
    if text[i] == "!" or text[i] == "." or text[i] == "?":
        sentences = sentences + 1
L = letter / space * 100
S = sentences / space * 100
index = 0.0588 * L - 0.296 * S - 15.8
if index >= 1 and index <= 16:
    print(f"Grade {index:.0f}")
elif index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")

File: CS50x/lecture2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");
    int letter = 0;
    int space = 1;
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        text[i] = toupper(text[i]);
        if (text[i] >= 'A' && text[i] <= 'Z')
        {
            letter++;
        }
    }
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            space++;
        }
    }
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sentences++;
        }
    }
    float L = (float) letter/space * 100;
    float S = (float) sentences/space * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index >= 1 && index <= 16)
    {
        printf("Grade %.0f\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

}
